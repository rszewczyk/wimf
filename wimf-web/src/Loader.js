// @flow
import React from "react";
import { css, merge } from "glamor";
import Spinner from "./Spinner";

const overlayStyle = css({
  position: "fixed",
  top: 0,
  left: 0,
  bottom: 0,
  right: 0,
  background: "rgba(0,0,0,.8)",
  display: "none",
  zIndex: 999
});

const overlayVisibleStyle = merge(overlayStyle, {
  display: "flex",
  alignItems: "center",
  justifyContent: "center"
});

type LoaderProps = {
  visible: boolean
};

export default function Loader(props: LoaderProps) {
  return (
    <div {...(props.visible ? overlayVisibleStyle : overlayStyle)}>
      <Spinner />
    </div>
  );
}
