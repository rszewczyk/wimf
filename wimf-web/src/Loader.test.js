import React from "react";
import renderer from "react-test-renderer";
import Loader from "./Loader";

it("renders correctly", () => {
  expect(renderer.create(<Loader />).toJSON()).toMatchSnapshot();
});
