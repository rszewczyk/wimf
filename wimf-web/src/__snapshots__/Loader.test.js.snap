// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-css-185on8x=""
>
  <div
    data-css-1ik39hp=""
    style={undefined}
  >
    <div />
  </div>
</div>
`;
